pkg/ns/ns.go

```diff
@@ -74,11 +75,22 @@ func GetCurrentNS() (NetNS, error) {
 
 // Returns an object representing the namespace referred to by @path
 func GetNS(nspath string) (NetNS, error) {
+	const NSFS_MAGIC = 0x6e736673
+
 	fd, err := os.Open(nspath)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("Failed to open %v: %v", nspath, err)
+	}
+
+	stat := syscall.Statfs_t{}
+	if err = syscall.Statfs(nspath, &stat); err != nil {
+		return nil, fmt.Errorf("Failed to Statfs %v: %v", nspath, err)
 	}
-	return &netNS{file: fd}, nil
+
+	return &netNS{
+		file:    fd,
+		mounted: stat.Type == NSFS_MAGIC,
```

Comment:
I wouldn't set mounted:true for plain GetNS() calls.  The  reason is because only the creator of the netns should be disposing of it via ns.Close(), and things that call GetNS() are the creator.  I think all you want to do here is return an error if the mountpoint is not an NSFS mount point.